client/product-details/index.js

```diff
@@ -1,6 +1,40 @@
+/* global jQuery */
+// global wcpayStripeSiteMessaging
+
 /**
  * Internal dependencies
  */
-import bnplSiteMessaging from './bnpl-site-messaging';
+import { initializeBnplSiteMessaging } from './bnpl-site-messaging';
+
+const bnplPaymentMessageElement = initializeBnplSiteMessaging();
+
+jQuery( function ( $ ) {
+	$( '.quantity input' ).on( 'change', function ( event ) {
+		const newQuantity = event.target.value;
+		const price = window.wcpayStripeSiteMessaging.price;
+		bnplPaymentMessageElement.update( {
+			amount: parseInt( price, 10 ) * newQuantity,
+		} );
+	} );
+
+	// Handle BNPL payment message changes for product variation.
+	$( '.single_variation_wrap' ).on( 'show_variation', function (
+		event,
+		variation
+	) {
+		const { variationsPriceList } = window.wcpayStripeSiteMessaging;
+		const variationPrice = variationsPriceList[ variation.variation_id ];
+		const quantity = $( '.quantity input' ).val();
+
+		window.wcpayStripeSiteMessaging.price = variationPrice;
+		bnplPaymentMessageElement.update( {
+			amount: parseInt( variationPrice, 10 ) * quantity,
+		} );
+	} );
 
-bnplSiteMessaging();
+	$( '.reset_variations' ).on( 'click', function () {
+		bnplPaymentMessageElement.update( {
+			amount: 0,
```

Comment:
Nope, we  should reset to the default product price. `const price = window.wcpayStripeSiteMessaging.price;`

Reply:
Or given proposal to always send array with prices, the first item from that list.

Reply:
Done!  The base product price is available with an index `base_product`

0129a74